library/alloc/src/vec/into_iter.rs

```diff
@@ -148,7 +148,7 @@ impl<T, A: Allocator> Iterator for IntoIter<T, A> {
 
     #[inline]
     fn next(&mut self) -> Option<T> {
-        if self.ptr as *const _ == self.end {
+        if self.ptr == self.end {
```

Comment:
I'd guess this was left from times when `self.ptr` and `self.end` had different types?... But they currently use the same type, so this is useless: https://github.com/rust-lang/rust/blob/de9da0ba8f6627e3aef4c01a95587041867e4625/library/alloc/src/vec/into_iter.rs#L41-L42